### PR TITLE
Use QUORUM consistency for reading migration information

### PR DESCRIPTION
**Pull request by elventear**
Following 188985a145428e32280d068090f038ffbc2e25ac (#12), reduce the consistency
for some some readonly operations during migration that were missing in
the previous commit.

### CODE REVIEW

File: src/main/java/smartthings/cassandra/CassandraConnection.java [hunk 1]
```diff
@@ -131,9 +131,17 @@ public ResultSet execute(Statement query) {
 		return session.execute(query);
 	}
 
+	public ResultSet executeWithQuorum(String query, Object... params) {
+		return executeWithQuorum(new SimpleStatement(query, params));
+	}
+
+	public ResultSet executeWithQuorum(Statement query) {
+		return execute(query.setConsistencyLevel(ConsistencyLevel.QUORUM));
```
Comment (arunrao33) [inline src/main/java/smartthings/cassandra/CassandraConnection.java]: make private

File: src/main/java/smartthings/cassandra/CassandraConnection.java [hunk 2]
```diff
@@ -172,13 +180,13 @@ public void setupMigration() {
 	public boolean migrationsTableExists() {
 		logger.debug("Checking for migrations table.");
 		if (cassandraVersion.startsWith("3.")) {
-			ResultSet existingMigration = execute("SELECT table_name " +
+			ResultSet existingMigration = executeWithQuorum("SELECT table_name " +
 							"FROM system_schema.tables	" +
 							"WHERE keyspace_name=? and table_name = 'migrations';",
```
Comment (arunrao33) [inline src/main/java/smartthings/cassandra/CassandraConnection.java]: Also, another question. This migration picks up the default consistency level from application's yaml file which by default is "Local_quorum", right?

Comment (elventear) [reply src/main/java/smartthings/cassandra/CassandraConnection.java]: AFAICS, the consistency level is set [here](https://github.com/SmartThingsOSS/cassandra-migration/pull/15/files#diff-0d110b466b4d9b8c4263831c889cb7aaL66), which is set to `ALL` and can't be overriden.

File: src/main/java/smartthings/cassandra/CassandraConnection.java (hunk 1 above)
Comment (elventear) [reply src/main/java/smartthings/cassandra/CassandraConnection.java]: done.

**Comment by beckje01** [comment]
Whats the reason we want to downgrade the consistency

**Comment by elventear** [comment]
@beckje01 we’ve run into situations when we are doing a new deployment that does not have any outstanding schema changes and the cassandra cluster was not fully healthy (1, or 2 nodes down) that would fail because the `ALL` consistency requirement for all the operations. The goal of these changes is to lower the consistency for reading the migration information, so a deployment can succeed even when all the nodes are available when there are no schema changes, which is most of the times.

When there are schema changes needed and the cluster is not fully available we still have to bite the bullet.

**Comment by arunrao33** [comment]
I'm still not very comfortable making schema changes when some nodes are down. I will let  @beckje01  make the  call.

**Comment by darksun** [comment]
Unless I missed something this wouldn't be making any schema changes when nodes are down.

**Comment by elventear** [comment]
@arunrao33 I agree. These changes will not change that.

**Comment by arunrao33** [comment]
We set the session with QueryOptions to consistency level "ALL". The first thing we do is to check migrationsTableExists() where we want to do it with Quorum but using the same session for subsequent operations. What i'm not sure is that if the consistency is changed for all the later queries. If it does, then we are running everything with quorum including schema changes. I might be wrong.. We should be able to test it quick though.

**Comment by beckje01** [comment]
So the need is deploys to work with an unhealthy cluster

**Comment by jeff-blaisdell** [comment]
👀

**Pull request closed without merging**